CMakeLists.txt

```diff
@@ -1,73 +1,71 @@
-cmake_minimum_required(VERSION 3.16 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.16 FATAL_ERROR) # Policies <= CMP0097 default to NEW
 
-message(AUTHOR_WARNING "If the build fails, please try the autotools/qmake method.")
-
-if (POLICY CMP0074)
-    cmake_policy(SET CMP0074 NEW)
-endif()
+set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules)
 
 # TODO: fix the macOS bundle stuff
-if(POLICY CMP0080)
-    cmake_policy(SET CMP0080 OLD)
+cmake_policy(SET CMP0080 OLD)
+
+# this must be defined before project() or enable_language() to take effect
+if(WIN32)
+    if(GENERATOR_IS_MULTI_CONFIG)
+        set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<BOOL:${QBT_MSVC_RUNTIME_DYNAMIC}>:DLL>$<$<CONFIG:Debug>:Debug>")
+    else()
+        set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<BOOL:${QBT_MSVC_RUNTIME_DYNAMIC}>:DLL>$<$<CMAKE_BUILD_TYPE:Debug>:Debug>")
+    endif()
 endif()
 
-list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules)
+# project version
 include(FunctionReadVersion)
+read_version("${CMAKE_CURRENT_SOURCE_DIR}/version.pri" QBT_VER_MAJOR QBT_VER_MINOR QBT_VER_BUGFIX QBT_VER_BUILD QBT_VER_STATUS)
+message(STATUS "Project version is: ${QBT_VER_MAJOR}.${QBT_VER_MINOR}.${QBT_VER_BUGFIX}.${QBT_VER_BUILD} (${QBT_VER_STATUS})")
 
-read_version("${CMAKE_CURRENT_SOURCE_DIR}/version.pri" VER_MAJOR VER_MINOR VER_BUGFIX VER_BUILD VER_STATUS)
-# message(STATUS "Project version is: ${VER_MAJOR}.${VER_MINOR}.${VER_BUGFIX}.${VER_BUILD} (${VER_STATUS})")
-
-project(qBittorrent VERSION ${VER_MAJOR}.${VER_MINOR}.${VER_BUGFIX}.${VER_BUILD})
+project(qBittorrent VERSION ${QBT_VER_MAJOR}.${QBT_VER_MINOR}.${QBT_VER_BUGFIX}.${QBT_VER_BUILD})
 
 # check for invalid compiler version/CXX standard as early as possible
+# ideally, we would also be able to check whether or not libtorrent was compiled
+# with the same standard from imported target properties
 include(FunctionQbtCXXCompilerAndModeCheck)
 qbt_minimum_cxx_mode_check(14)
-message(STATUS "Building in C++${CMAKE_CXX_STANDARD} mode.\n"
-            "Make sure libtorrent was built with the same C++ mode for ABI compatibility.")
-
-set(PROJECT_VERSION "${VER_MAJOR}.${VER_MINOR}.${VER_BUGFIX}")
-
-if (NOT VER_BUILD EQUAL 0)
-    set(PROJECT_VERSION "${PROJECT_VERSION}.${VER_BUILD}")
-endif()
-
-set(PROJECT_VERSION "${PROJECT_VERSION}${VER_STATUS}")
-
-add_definitions(-DQBT_VERSION_MAJOR=${VER_MAJOR})
-add_definitions(-DQBT_VERSION_MINOR=${VER_MINOR})
-add_definitions(-DQBT_VERSION_BUGFIX=${VER_BUGFIX})
-add_definitions(-DQBT_VERSION_BUILD=${VER_BUILD})
-
-add_definitions(-DQBT_VERSION="v${PROJECT_VERSION}")
-add_definitions(-DQBT_VERSION_2="${PROJECT_VERSION}")
-
-include(GNUInstallDirs)
-include(FeatureSummary)
 
 # version requirements
-set(requiredBoostVersion 1.40)
+set(requiredBoostVersion 1.60)
 set(requiredQtVersion 5.9.0)
 set(requiredOpensslVersion 1.0)
-set(requiredLibtorrentVersion 1.1.10)
+set(requiredLibtorrentVersion 1.2.0)
```

Comment:
I think we still don't agree we should drop it and you should leave these decisions to maintainers, they will drop it themselves when they think they will not maintain it. so please don't argue with this and also the boost maintainers, these core library decision should be left on those who actually maintain it

Reply:
Like I said in the OP:
> Revise and possibly undo some dependency version bumps that for now are just here to simplify things.

I was running into problems with vcpkg rejecting packages with incompatible minor versions. This was the simplest way I found to "fix" the problem for now, while I focus on everything else. Once other things are ready, this will be revisited, and I'll see what are the minimum set of changes required to support both series if needed.

Reply:
Seems the version numbers still hasn't reverted.

Reply:
@glassez 
Do you agree with this change?
https://github.com/qbittorrent/qBittorrent/pull/12746/files#r419148054
https://github.com/qbittorrent/qBittorrent/pull/12746/files#r449756807

Reply:
@Chocobo1 @glassez

The landscape has changed a bit since this PR was opened, most notably, support for `RC_1_1` was officially dropped from master.

As for the remaining dependencies: Ubuntu is basically our lower bound, and the minimum requirement of Qt 5.9 means it also does not make sense to _not_ bump versions up to 18.04 versions, e.g. zlib up to 1.2.11, boost up to 1.65 and OpenSLL up to 1.1.1 (plus OpenSSL 1.1.1 is actually mandatory in Qt >= 5.15, so at least this bump makes things easier).

Reply:
>so at least this bump makes things easier).

But your opinion doesn't really comply to what we established before (before you join) and I still see no point of your reasoning. I don't have much free time and I would just say this much.

Reply:
If you want us to seriously consider CMake as something approaching the official build system, you must follow the official project requirements. If you want to change them you should open separate Issue. Usually we discuss it before upcoming minor/major release.